[ejnan] iP

ejnan opened the pull request:
# Alpa

Alpa is your fluffy task manager that frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/ejnan/ip/releases/download/A-Release/alpa.jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Find tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ejnan committed:
Customise chatbot with name and greetings

ejnan committed:
Echo user input and format interface

ejnan committed:
Store user input as a list and display when requested

ejnan committed:
Add Task.java to represent class and chatbot's ability to mark tasks

ejnan committed:
Support to track ToDos Events and Deadlines

ejnan committed:
Refactor code to handle different errors

ejnan committed:
Add support for deleting tasks

ejnan committed:
Set up files for automated text UI testing

ejnan committed:
Implement enum for task types

ejnan committed:
Implement Save to and Load from hard disk feature

ejnan committed:
Merge branch-Level-7 into master

jiahui0309 commented on src/main/java/Alpa.java:
```diff
@@ -0,0 +1,284 @@
+    printDecoratedMessage(listOutput.toString());
+  }
+
+  private static void handleMarkUnmark(String[] parts, CommandType command) throws AlpaException {
```
Perhaps you can consider naming it `handleMark` instead to make it less redundant, while having sufficient details.

jiahui0309 commented on src/main/java/TaskType.java:
```diff
@@ -0,0 +1,31 @@
+  DEADLINE,
+  EVENT;
+
+  public String getShortName() {
```
The naming of this method was a little confusing for me. Perhaps you can consider names such as `getInitials()`?

jiahui0309 commented on src/main/java/CommandType.java:
```diff
@@ -0,0 +1,12 @@
+public enum CommandType {
+    BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, INVALID;
```
I like how the names of the command types are short and easy to understand.

jiahui0309 commented on src/main/java/Alpa.java:
```diff
@@ -0,0 +1,284 @@
+    }
+  }
+
+  private static void saveTasks() {
```
I like how you clearly distinguish between single-valued and multi-valued variables throughout your code files.

jiahui0309 commented on src/main/java/Alpa.java:
```diff
@@ -0,0 +1,284 @@
+    System.out.println("𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣\n");
+  }
+
+  private static void handleList() {
+    if (tasks.isEmpty()) {
+      printDecoratedMessage("\nYour list is empty, human!");
+      return;
+    }
+    StringBuilder listOutput = new StringBuilder("\nYour list, human!\n");
+    for (int i = 0; i < tasks.size(); i++) {
+      listOutput.append("  ").append(i + 1).append(". ").append(tasks.get(i)).append("\n");
+    }
+    printDecoratedMessage(listOutput.toString());
+  }
+
```
I like how the namings are intuitive, making it easier for me to understand the method that you are calling.

jiahui0309 reviewed (commented):
LGTM! Overall, your methods and classes are well-named. It was not misleading and intuitive for me. I also like how you named related methods similarly such as `handleEvent` and `handleDeadline`. :)

KohGuanZeh commented on src/main/java/Alpa.java:
```diff
@@ -0,0 +1,284 @@
+            handleList();
+            break;
+        
+          case MARK:
```
I think that maybe a comment would be good to indicate that the fall through is intentional?

KohGuanZeh commented on src/main/java/Alpa.java:
```diff
@@ -0,0 +1,284 @@
+        String[] parts = input.split(" ");
+        CommandType command = CommandType.fromString(parts[0]);
+
+        switch (command) {
```
I think that `case` keywords should align with the switch statement?

KohGuanZeh commented on src/main/java/Alpa.java:
```diff
@@ -0,0 +1,284 @@
+    tasks.add(task);
+    saveTasks();
+    StringBuilder message = new StringBuilder();
+    message.append("\nYou added a task human!\n" + "  " + task).append("\n")
+           .append("Now you have ").append(tasks.size()).append(" tasks in your list!");
```
I like how you break the lines which helped to maintain readability.

KohGuanZeh commented on src/main/java/Alpa.java:
```diff
@@ -0,0 +1,284 @@
+import java.util.Scanner;
+
+public class Alpa {
+  private static final List<Task> tasks = new ArrayList<>();
```
Perhaps indentations should be 4 spaces instead of 2?

KohGuanZeh commented on src/main/java/TaskType.java:
```diff
@@ -0,0 +1,31 @@
+  EVENT;
+
+  public String getShortName() {
+    switch (this) {
+      case TODO:
+        return "T";
+      case DEADLINE:
+        return "D";
+      case EVENT:
+        return "E";
+      default:
+        return "Unknown";
```
Maybe `case` keyword is meant to align with switch statement?

KohGuanZeh reviewed (commented):
I have noticed that you have put a lot of effort into developing Duke to be something special to you and I like the personality of the bot you have made. I did notice a few issues with regards to the given Java conventions and I think you have missed out (styling of `switch` `case` in some areas). Perhaps you can look into your IDE settings and configure them to follow the respective guidelines and it will help to automatically reformat the code on save which will save time when refactoring. Overall great job and hope to see more progress. 😄

Darylgolden commented on src/main/java/Alpa.java:
```diff
@@ -0,0 +1,284 @@
+    System.out.println("I'm here to help you sort through the woolly world of tasks.\n");
+    System.out.println("෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘෴෴⚘\n");
+    
+    boolean exit = false;
```
I think the variable name here should be phrased to sound like a boolean?

Darylgolden commented on src/main/java/Alpa.java:
```diff
@@ -0,0 +1,284 @@
+    }
+  }
+
+  private static void handleBye() {
```
I like the consistent naming scheme of `handle_action` throughout here.

Darylgolden commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,36 @@
+  }
+
+  public String toFileFormat() {
+    String status = isDone ? "1" : "0";
```
Perhaps the naming here could be a bit more explicit?

Darylgolden reviewed (commented):
Other than a few small lapses, I think you did a good job keeping to good naming standards in your code!

ejnan committed:
Allow Alpa to understand date and time inputs

ejnan committed:
Allow Alpa to accept more types of inputs

ejnan committed:
Merge branch-Level-8 into master

ejnan committed:
Reformat code to make it more OOP

ejnan committed:
Merge A-MoreOOP into master

ejnan committed:
Organise classes into packages

ejnan committed:
Merge A-Packages into master

ejnan committed:
Merge branch 'add-gradle-support'

ejnan committed:
Add Gradle support

ejnan committed:
Clean up files

ejnan committed:
Add JUnit tests

ejnan committed:
Merge branch-A-JUnit into master

ejnan committed:
Add JavaDoc comments

ejnan committed:
Comply with coding standards

ejnan committed:
Implement Level-9 Find feature

ejnan committed:
Merge branch 'branch-A-CodingStandard'

ejnan committed:
Merge branch 'branch-Level-9'

ejnan committed:
Use CheckStyle to detect coding style violations

ejnan committed:
Merge branch-A-CheckStyle into master

ejnan committed:
Add GUI to Alpa task manager chatbot

ejnan committed:
Merge branch-Level-10

ejnan committed:
Add assertions to Deadline and Event constructors

Both Deadline and Event classes rely on non-null and logically consistent
constructor parameters to function correctly. However, there were no
preconditions checks for these parameters.

To prevent such scenarios and enforce data integrity right from object
creation, assertions have been added to the constructors of both classes.

- In Deadline, assertions ensure the description and deadline are non-null
  and that the deadline is a future date.
- In Event, assertions check for non-null descriptions, start and end
  times, and additionally, that the end time is after the start time.

These checks are crucial for maintaining the internal consistency of task
objects and preventing errors due to invalid data. Assertions were chosen
over other validation mechanisms as they serve as both documentation and
runtime checks without affecting the production code's performance.

ejnan committed:
Refactor Storage.java and Enhance Parser.java for Better Code Quality

The refactoring of Storage.java addresses the need for clearer, more
isolated logic for task parsing, which is crucial for the scalability
and maintainability of the code. The addition of comments in Parser.java
fulfills the goal of making the code more accessible and understandable
for future enhancements.

Key Changes:
- Storage.java:
  * Extracted task parsing logic into `parseToDoTask`, `parseDeadlineTask`,
    and `parseEventTask` methods.
  * Improved error handling by adding more descriptive error messages,
    aiding in quicker identification of issues with task file format.
  * Ensured directory existence check and creation logic is more robust,
    preventing potential issues with missing directories.

- Parser.java:
  * Added detailed comments to the `parse` method, explaining the logic
    behind command parsing and handling of different types of tasks.
  * Clarified the role of each case in the switch statement, providing
    insights into how various commands are processed.

ejnan committed:
Merge pull request #3 from ejnan/A-Assertions

Add assertions to Deadline and Event constructors

ejnan committed:
Merge pull request #2 from ejnan/A-CodeQuality

Refactor Storage.java and Enhance Parser.java for Better Code Quality

ejnan committed:
Setup GitHub Actions for Java CI

ejnan committed:
Merge pull request #4 from ejnan/branch-A-CI

Setup GitHub Actions for Java CI

ejnan committed:
Implement Extension C-Help

ejnan committed:
Merge branch-BCD-Extension into master

ejnan committed:
Update gradle.yml

ejnan committed:
Enhance UI Responsiveness and Aesthetics

The GUI has been updated to improve responsiveness and visual appeal.

Add style.css to remove size restrictions on dialog boxes and
their content, and added styles for spacing and padding to
improve readability and overall aesthetics of the dialog boxes.

ejnan committed:
Merge branch-A-BetterGui into master

ejnan committed:
Update GUI to improve formatting and resizeability

ejnan committed:
Add User Guide to README.md

ejnan committed:
Merge branch-A-UserGuide into master

ejnan committed:
Edit example in README.md for adding Event
